ProjectTemplates/templates/Functions-CSharp/.template.config/template.json

```diff
@@ -0,0 +1,232 @@
+            ],
+            "replaces": "net5.0",
+            "defaultValue": "net5.0"
+        },
+        "copyrightYear": {
+            "type": "generated",
+            "generator": "now",
+            "replaces": "copyrightYear",
+            "parameters": {
+                "format": "yyyy"
+            }
+        },
+        "skipRestore": {
+            "type": "parameter",
+            "datatype": "bool",
+            "description": "If specified, skips the automatic restore of the project on create.",
+            "defaultValue": "false"
+        },
+        "CalledApiUrl": {
+            "type": "parameter",
+            "datatype": "string",
+            "replaces": "[WebApiUrl]",
+            "defaultValue": "https://graph.microsoft.com/v1.0",
+            "description": "URL of the API to call from the web app. This option only applies if --auth SingleOrg or --auth IndividualB2C is specified."
+        },
+        "CallsMicrosoftGraph": {
+            "type": "parameter",
+            "datatype": "bool",
+            "defaultValue": "false",
+            "description": "Specifies if the web app calls Microsoft Graph. This option only applies if --auth SingleOrg is specified."
+        },
+        "CalledApiScopes": {
+            "type": "parameter",
+            "datatype": "string",
+            "replaces": "user.read",
+            "description": "Scopes to request to call the API from the web app. This option only applies if --auth SingleOrg or --auth IndividualB2C is specified."
+        },
+        "GenerateApi": {
+            "type": "computed",
+            "value": "((IndividualB2CAuth || OrganizationalAuth) && (CalledApiUrl != \"https://graph.microsoft.com/v1.0\" || CalledApiScopes != \"user.read\"))"
+        },
+        "GenerateGraph": {
+            "type": "computed",
+            "value": "(OrganizationalAuth && CallsMicrosoftGraph)"
+        },
+        "GenerateApiOrGraph": {
+            "type": "computed",
+            "value": "(GenerateApi || GenerateGraph)"
+        }
+    },
+    "StorageConnectionStringValue": {
+        "description": "The connection string for your Azure WebJobs Storage.",
+        "type": "parameter",
+        "defaultValue": "UseDevelopmentStorage=true",
+        "replaces": "AzureWebJobsStorageConnectionStringValue",
+        "DataType": "AzureStorage"
+    },
+    "AzureFunctionsVersion": {
+        "defaultValue": "V2",
+        "description": "The setting that determines the target release",
+        "replaces": "AzureFunctionsVersionValue",
+        "type": "parameter"
+    },
+    "sources": [
+        {
+            "modifiers": [
+                {
+                    "condition": "(ExcludeLaunchSettings)",
+                    "exclude": [
+                        "Properties/launchSettings.json"
+                    ]
+                },
+                {
+                    "condition": "(NoAuth)",
+                    "exclude": [
+                        "Startup.cs"
```

Comment:
We'll also want to exclude the `appsettings.json` (which is added by the `Startup.cs`)